p2p/src/sync/mod.rs

```diff
@@ -47,6 +47,10 @@ const HEADER_LIMIT: usize = 2000;
 // TODO: split syncing into separate files
 // TODO: match against error in `run()` and deal with `ProtocolError`
 // TODO: use ensure
+// TODO: create better api for request/response codec
+// TODO: create helper function for creating requests/responses
+// TODO: cache locator and invalidate it when `NewTip` event is received
+// TODO: simplify code
```

Comment:
Maybe more details on "simplify code"... we tend to forget the reason for thinking the code is "complex".

Reply:
I'll expand the comment but it's impossible for me to forget that it means because you just need to take a look at the current code and see how bad it is to know what to simplify.